src/styles/withTheme.js

```diff
@@ -18,15 +19,18 @@ function getDefaultTheme() {
   return defaultTheme;
 }
 
+type InjectedProps = { theme: Object };
+
 // Provide the theme object as a property to the input component.
-export default function withTheme<BaseProps: {}>(BaseComponent: ComponentType<BaseProps>) {
-  const factory = createEagerFactory(BaseComponent);
+const withTheme = (): HigherOrderComponent<{}, InjectedProps> => (Component: any): any => {
```

Comment:
I don't get it. Why changing the API?

Reply:
It's unlikely that we will every need configuration options.

Reply:
Consistency. Sanity. My sanity.  Seriously. If you want to edit it back, you can, but I'd just prefer one chosen HOC pattern.

Reply:
Maybe @istarkov could give us some light here. For instance, I can see the `pure` typing.

https://github.com/acdlite/recompose/blob/6064554ccceb07d126f9df61d4d6221c5fc726ea/types/flow-typed/recompose_v0.24.x/flow_v0.55.x-/recompose_v0.24.x.js#L138

@rosskevin Are you saying that we need `HigherOrderComponent` to get the flow going through the HOCs?

Reply:
@rosskevin recompose seem to have HOC factories as well as HOCs depending on the use cases. I will work on this if you prefer 👼 .

Reply:
`HigherOrderComponent` solves the default prop not found issue linked in the description.

Reply:
Yep, for most enhancers to type them properly they must be a function, pure is bad example as it just do nothing. So based on my experience there are no other ways